sdk/ml/azure-ai-ml/azure/ai/ml/operations/_capability_hosts_operations.py

```diff
@@ -0,0 +1,345 @@
+
+    def __validate_properties(self, capability_host: CapabilityHost, workspace_kind: str) -> None:
+        """Validate the properties of the capability host based on the workspace kind.
+
+        :param capability_host: The capability host to validate.
+        :type capability_host: CapabilityHost
+        :param workspace_kind: The kind of the workspace, either hub or project only.
+        :type workspace_kind: str
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if workspace kind is not Hub or Project.
+            Details will be provided in the error message.
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if the OpenAI service connection or
+            vector store (AISearch) connection is empty for a Project workspace kind.
+            Details will be provided in the error message.
+        :return: None, or the result of cls(response)
+        :rtype: None
+        """
+
+        if not self.__is_valid_kind(workspace_kind):
+            msg = f"Invalid workspace kind: {workspace_kind}. Workspace kind should be either 'Hub' or 'Project'."
+            raise ValidationException(
+                message=msg,
+                target=ErrorTarget.CAPABILITY_HOST,
+                no_personal_data_message=msg,
+                error_category=ErrorCategory.USER_ERROR,
+            )
+
+        if workspace_kind == WorkspaceKind.PROJECT:
+            if capability_host.ai_services_connections is None or capability_host.vector_store_connections is None:
+                msg = "For Project workspace kind, OpenAI service connections and vector store (AISearch) connections are required."  # pylint: disable=line-too-long
+                raise ValidationException(
+                    message=msg,
+                    target=ErrorTarget.CAPABILITY_HOST,
+                    no_personal_data_message=msg,
+                    error_category=ErrorCategory.USER_ERROR,
+                )
+
+    def __is_valid_kind(self, kind: str):
+        """Validate if the provided kind is a valid WorkspaceKind.
+
+        :param kind: The kind of workspace to validate, either hub or project only.
+        :type kind: str
+        :return: True if the kind is valid, False otherwise.
+        :rtype: bool
+        """
+
+        return kind in {WorkspaceKind.HUB, WorkspaceKind.PROJECT}
+
+    def __get_workspace(self) -> Workspace:
+        """Retrieve the workspace object.
+
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if specified Hub or Project do not exist.
+            Details will be provided in the error message.
+        :return: Hub or Project object if it exists
+        :rtype: ~azure.ai.ml.entities._workspace.workspace.Workspace
+        """
+        rest_workspace = self._workspace_operations.get(self._resource_group_name, self._workspace_name)
+        workspace = Workspace._from_rest_object(rest_workspace)
+        if workspace is None:
```

Comment:
this never gets hit since `get` will raise an exception if the workspace does not exist.

Reply:
we need this function call for retrieving the kind of workspace. and workspace._from_rest_object returning the None hence, its throwing an error while returning None or workspace where as we always need Workspace.

Reply:
When do we ever get back a `None` workspace? If it doesn't exist, we should be getting a 404 when calling GET on the workspace

Reply:
as per discussion over the teams call, this change is required.